### PR TITLE
#1465 Make checkbox component available for View Editor

### PR DESCRIPTION
**Pull request by emil-botha**
- Create a CheckboxField component for form input fields
- Change the Checkbox component to receive a ref prop

### CODE REVIEW

File: packages/bappo-components/package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "bappo-components",
-  "version": "0.1.0-alpha.186",
+  "version": "0.1.0-alpha.187",
```
Comment (williamfeng91) [inline packages/bappo-components/package.json]: Hi Emil, we normally update the version when we publish and make it a commit of itself so we can easily locate the version changes when we look at git history. Jon has written a great documentation for the publish process: https://app.gitbook.com/@bappo/s/bappo-courses/bappo-component/develop-components-and-test-them#6-publish

File: packages/bappo-components/src/primitives/Checkbox/types.ts [hunk 2]
```diff
@@ -8,7 +8,7 @@ export type CheckboxProps = {
   /**
    * The checked state of the Checkbox. If true the Checkbox will be checked. Default value is false.
    */
-  checked?: boolean;
+  checked?: boolean | null;
```
Comment (williamfeng91) [inline packages/bappo-components/src/primitives/Checkbox/types.ts]: I wonder if we should use `value` to be consistent with all other inputs?

File: packages/bappo-components/src/primitives/Checkbox/Checkbox.native/Checkbox.tsx [hunk 3]
```diff
@@ -7,85 +7,105 @@ import {
...
+    const _onKeyPress = (event: React.KeyboardEvent<TouchableOpacity>) => {
+      const ENTER = 13;
+      const SPACE = 32;
+
+      if (event.which === ENTER || event.which === SPACE) {
+        event.preventDefault();
+        event.stopPropagation();
+        _toggle();
+      }
+    };
```
Comment (williamfeng91) [inline packages/bappo-components/src/primitives/Checkbox/Checkbox.native/Checkbox.tsx]: Do we need this for native?

**Review by williamfeng91 — CHANGES_REQUESTED** [review]

File: packages/bappo-components/package.json (hunk 1 above)
Comment (emil-botha) [reply packages/bappo-components/package.json]: Hi William, thanks for referring me to Jon's write-up. I've reverted the version change.

File: packages/bappo-components/src/primitives/Checkbox/Checkbox.native/Checkbox.tsx (hunk 3 above)
Comment (emil-botha) [reply packages/bappo-components/src/primitives/Checkbox/Checkbox.native/Checkbox.tsx]: The `TouchableOpacity` component allows for keyboard input so I kept it to be consistent with web but I also think it is unnecessary and removed it.

**Pull request merged**